File: Modules/ServerTypes/WorkFlow/RuleForm.ts
import { BooleanEditor, initFormType, LookupEditor, PrefixedContext, ServiceLookupEditor } from "@serenity-is/corelib";

export interface RuleForm {
    RoleId: LookupEditor;
    CurrentStatusId: ServiceLookupEditor;
    ActionId: ServiceLookupEditor;
    NextStatusId: ServiceLookupEditor;
    IsFinalState: BooleanEditor;
}

export class RuleForm extends PrefixedContext {
    static readonly formKey = 'WorkFlow.Rule';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!RuleForm.init) {
            RuleForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;
            var w2 = BooleanEditor;

            initFormType(RuleForm, [
                'RoleId', w0,
                'CurrentStatusId', w1,
                'ActionId', w1,
                'NextStatusId', w1,
                'IsFinalState', w2
            ]);
        }
    }
}

File: Modules/WorkFlow/Rule/RuleForm.cs
namespace WorkFlow.Forms;

[FormScript("WorkFlow.Rule")]
[BasedOnRow(typeof(RuleRow), CheckNames = true)]
public class RuleForm
{
    public int RoleId { get; set; }
    public int CurrentStatusId { get; set; }
    public int ActionId { get; set; }
    public int NextStatusId { get; set; }    
    public bool IsFinalState { get; set; }
}